src/enforester.js

```diff
@@ -1326,25 +1328,20 @@ export class Enforester {
 
   enforestNewExpression() {
     this.matchKeyword('new');
-    let callee;
     let lookahead = this.peek();
-    if (this.isKeyword(lookahead, 'new')) {
-      callee = this.enforestNewExpression();
-    } else if (this.isKeyword(lookahead, 'super')) {
-      callee = this.enforestLeftHandSideExpression({ allowCall: false });
-    } else if (this.isParens(lookahead) || this.isBrackets(lookahead) || this.isBraces(lookahead) || this.isTemplate(lookahead)) {
-      callee = this.enforestExpressionLoop();
-    } else if (this.isPunctuator(lookahead, '.') && this.isIdentifier(this.peek(1), 'target')) {
+    if (this.isPunctuator(lookahead, '.') && this.isIdentifier(this.peek(1), 'target')) {
       this.advance();
       this.advance();
       return new Term('NewTargetExpression', {});
+    } else if(this.isIdentifier(lookahead)){
+      this.term = new Term('IdentifierExpression', { name : this.enforestIdentifier() });
     } else {
-      callee = new Term('IdentifierExpression', { name : this.enforestIdentifier() });
+      this.term = this.enforestLeftHandSideExpression({ allowCall: false });
     }
 
+    // (new a``) is a NewExpression while (new a()``) is a TemplateExpression
     if(this.isTemplate(this.peek())) {
-      this.term = callee;
-      callee = this.enforestLeftHandSideExpression({ allowCall: false });
+      this.term = this.enforestLeftHandSideExpression({ allowCall: false });
     }
```

Comment:
@disnet: You were right. It's cleaner. I think I'm still stuck with 1343-1345 though, unless you have some ideas. I guess I should hold off on responding until I have my :coffee: